packages/hash/frontend/src/pages/[account-slug]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/data-type-selector/custom-expected-value-builder/object-expected-value-builder.tsx

```diff
@@ -0,0 +1,308 @@
+import { VersionedUri } from "@blockprotocol/type-system";
+import { faAsterisk } from "@fortawesome/free-solid-svg-icons";
+import { Chip, FontAwesomeIcon } from "@hashintel/hash-design-system";
+import {
+  Box,
+  Checkbox,
+  chipClasses,
+  Stack,
+  Table,
+  TableBody,
+  TableCell,
+  TableCellProps,
+  TableHead,
+  TableRow,
+  Typography,
+} from "@mui/material";
+import { usePopupState } from "material-ui-popup-state/hooks";
+import { FunctionComponent, useEffect, useMemo, useState } from "react";
+import { useFormContext, useWatch } from "react-hook-form";
+import { usePropertyTypesContextValue } from "../../../../../../use-property-types-context-value";
+import { PropertyTypeFormValues } from "../../../property-type-form-values";
+import { DeleteExpectedValueModal } from "./array-expected-value-builder/delete-expected-value-modal";
+import { ExpectedValueBadge } from "./array-expected-value-builder/expected-value-badge";
+import { CustomExpectedValueSelector } from "./shared/custom-expected-value-selector";
+
+const StyledTableHeadCell = (props: TableCellProps) => (
+  <TableCell
+    {...props}
+    sx={{
+      py: 1,
+      px: 1.5,
+      lineHeight: "18px",
+      fontSize: 11,
+      fontWeight: 600,
+      borderBottom: ({ palette }) => `1px solid ${palette.gray[30]}`,
+      whiteSpace: "nowrap",
+    }}
+  />
+);
+
+const StyledTableBodyCell = (props: TableCellProps) => (
+  <TableCell
+    {...props}
+    sx={{
+      py: 1,
+      px: 1.5,
+      flexDirection: "row",
+      alignItems: "center",
+      borderBottom: ({ palette }) => `1px solid ${palette.gray[30]} !important`,
+    }}
+  />
+);
+
+export interface ObjectExpectedValueRowProps {
+  objectId: string;
+  propertyId: string;
```

Comment:
Why not typed as `VersionedUri`?

Reply:
Done in b830bf2cba7cfed9795d4decdfa264d33e6befa3